x/finality/keeper/msg_server.go

```diff
@@ -53,9 +53,12 @@ func (ms msgServer) AddFinalitySig(goCtx context.Context, req *types.MsgAddFinal
 		return nil, err
 	}
 
-	fpPK := req.FpBtcPk
-
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	if err := ms.validateActivationHeightAddFinalitySig(ctx, req); err != nil {
```

Comment:
I think we should return here https://github.com/babylonlabs-io/babylon/blob/dd108f4013d9de0e684336134f49c14bd8eac2d1/x/btcstaking/keeper/power_dist_change.go#L52 if the finality is not activated. This way, voting power is not assigned after delegation events are processed so that we won't start tracking voting history of a newly active fp